processor/processorhelper/processorhelperprofiles/profiles.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package processorhelperprofiles // import "go.opentelemetry.io/collector/processor/processorhelper/processorhelperprofiles"
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/consumerprofiles"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.opentelemetry.io/collector/processor/processorprofiles"
+)
+
+// ProcessProfilesFunc is a helper function that processes the incoming data and returns the data to be sent to the next component.
+// If error is returned then returned data are ignored. It MUST not call the next component.
+type ProcessProfilesFunc func(context.Context, pprofile.Profiles) (pprofile.Profiles, error)
+
+type profiles struct {
+	component.StartFunc
+	component.ShutdownFunc
+	consumerprofiles.Profiles
+}
+
+// NewProfiles creates a processorprofiles.Profiles that ensure context propagation and the right tags are set.
```

Comment:
This processor currently doesn't set any "tags", it only provides a helper and context propagation.

```suggestion
// NewProfiles creates a processorprofiles.Profiles that ensure context propagation.
```